### PR TITLE
Adds the terraform provider page

### PR DESCRIPTION
**Pull request by ecomba**
It's a little walkthrough the configuration of our terraform provider

### CODE REVIEW

File: content/articles/terraform-provider.markdown [hunk 1]
```diff
@@ -0,0 +1,120 @@
+---
+title: Terraform Provider
+excerpt: How to use DNSimple's Terraform provider to manage your DNS
+categories:
+- Services
```
Comment (dallasread) [inline content/articles/terraform-provider.markdown:5]: All of the other articles in the Services category seem to relate to our one-click services. I'm not sure this one fits. 😄 

```suggestion
```

File: content/articles/terraform-provider.markdown [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+---
+
+[Terraform](https://terraform.io) is an Infrastructure as Code (IoC) tool that allows you to build, change, and version
+infrastructure safely and efficiently. With [DNSimple's Terraform Provider](https://registry.terraform.io/providers/dnsimple/dnsimple/latest),
+you can easily manage your DNS infrastructure and set up the required DNS records.
+
+## Configuring the DNSimple provider
+
+```
```
Comment (dallasread) [inline content/articles/terraform-provider.markdown:24]: I wish we had syntax highlighting!

File: content/articles/terraform-provider.markdown [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+When setting up a record (an MX (mail exchange) record in this example), provide the following information:
+
+    –	The zone name this record is for
+    –	A name (for your own records)
+    –	The value of the record (for the MX record in the example, this is the address to the mail server.)
+    –	The type of record (MX in the example) – see [Supported Record Types](https://support.dnsimple.com/articles/supported-dns-records/) for a comprehensive list of the records we support.
+    –	The time to live (TTL) of the record
+    –	The priority
```
Comment (dallasread) [inline content/articles/terraform-provider.markdown:120]: Could we have a final sentence/paragraph that includes our Terraform link again?

File: content/articles/terraform-provider.markdown [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+[API Access Token](https://support.dnsimple.com/articles/api-access-token/) support article for more information on how to get a token. 
+
+The **account** argument is the account ID associated with the token. For more information on obtaining your 
+account ID, take a look at [Account scope](https://developer.dnsimple.com/v2/#account-scope) in our 
+[developer documentation](http://developer.dnsimple.com).
+
+The **sandbox** argument tells the provider to use the sandbox environment. The token and account ID will vary 
+between the sandbox and our production environment. The sandbox environment is a great place to make sure you have your 
+configuration dialed in before using it in the actual production environment. To learn more about our sandbox 
+environment, read [Sandbox for Testing Calls to the API](https://support.dnsimple.com/articles/sandbox/).
```
Comment (dallasread) [inline content/articles/terraform-provider.markdown:46]: There's a few more cases similar to this with odd line-breaks.

What did you use to write this, Emacs? :trollface:

```suggestion
The **sandbox** argument tells the provider to use the sandbox environment. The token and account ID will vary between the sandbox and our production environment. The sandbox environment is a great place to make sure you have your configuration dialed in before using it in the actual production environment. To learn more about our sandbox environment, read [Sandbox for Testing Calls to the API](https://support.dnsimple.com/articles/sandbox/).
```

**Review by dallasread — APPROVED** [review]
This is great – I feel like I know how to use Terraform now!

Could we provide a complete list of the data sources that we support (or perhaps they are all here already)?

File: content/articles/terraform-provider.markdown (hunk 4 above)
Comment (ecomba) [reply content/articles/terraform-provider.markdown:45]: > 

Just vim... :trollface:

File: content/articles/terraform-provider.markdown (hunk 2 above)
Comment (sbastn) [reply content/articles/terraform-provider.markdown:23]: We do have the gem for it: `kramdown-parser-gfm` but it's true that it's not working at the moment. I'll open an issue to track this down.

Comment (sbastn) [reply content/articles/terraform-provider.markdown:23]: Nevermind, it was nanoc not refreshing the page correctly, but I got it to work:
Screen Shot 2021-11-30 at 10 02 56 AM

Here's how I did that

Screen Shot 2021-11-30 at 10 03 50 AM

However, it's true that the washed out red background is annoying. I'll fix that later.

File: content/articles/terraform-provider.markdown (hunk 3 above)
Comment (itsalyse) [reply content/articles/terraform-provider.markdown:116]: ✅

**Review by itsalyse — APPROVED** [review]

**Pull request merged**